### PR TITLE
Feature/onboarding wizard cleanup

### PR DESCRIPTION
**Pull request by agduncan94**
Onboarding wizard
-now uses material stepper
-first step allows users to change username if possible
-some rewording and cleanup for simplification

https://github.com/ga4gh/dockstore/issues/1691

Also
https://github.com/ga4gh/dockstore/issues/1690
https://github.com/ga4gh/dockstore/issues/1693

### CODE REVIEW

File: src/app/loginComponents/accounts/external/accounts.component.html [hunk 1]
```diff
@@ -1,44 +1,50 @@
...
-          </button>
-          <span class="form-inline" *ngIf="row.show">
-            <span class="input-group">
-              <!-- TODO: Only have 1 pipe here -->
-              <span class="form-control input-xs">{{row.source | getTokenContent: tokens}}</span>
+  <mat-card class="mb-2">
+    <div>
+      <h4>
+        <mat-chip-list>
+          <mat-chip *ngIf="row.isLinked" color="accent" selected matTooltip="Account is linked" [matTooltipShowDelay]="1000">
```
Comment (garyluu) [inline src/app/loginComponents/accounts/external/accounts.component.html]: can remove all matTooltipShowDelay unless it's the intention for it to be special

File: src/app/loginComponents/accounts/internal/change-username/change-username.component.html [hunk 2]
```diff
@@ -0,0 +1,29 @@
+<p *ngIf="showText">
+  Your current username is <strong>{{user?.username}}</strong>. It may have been auto-generated when you initially created a Dockstore account.
+  You can update your username only if you meet certain conditions.
+</p>
+
+<p>You <strong>can<span *ngIf="!extendedUser?.canChangeUsername">'t</span></strong> change your username.</p>
+
+<div class="change-username-form">
+  <mat-form-field>
+    <input matInput placeholder="Username" name="username" [(value)]="username" (keyup)="checkIfUsernameExists($event)" id="username">
```
Comment (garyluu) [inline src/app/loginComponents/accounts/internal/change-username/change-username.component.html]: change from keyup to something with debounceTime, search formInputDebounceTime for example.  Or, put a TODO

File: cypress/integration/dropdown.js [hunk 3]
```diff
@@ -75,22 +117,32 @@ describe('Dropdown test', function() {
             // })
             // TODO: Gitlab is being very slow, hopefully one day we can remove this
             // cy.wait(10000);
```
[the remark this answers is not in the document]
Comment (agduncan94) [reply cypress/integration/dropdown.js]: They will need to be slightly modified when we do enable gitlab. I'll make some comments.

File: src/app/navbar/navbar.component.css [hunk 4]
```diff
@@ -42,9 +49,7 @@
 }
 
 .navbar-holder {
-  margin: 0px !important;
-  padding-left: 15px;
-  padding-right: 15px;
+  width: 100% !important;
```
Comment (garyluu) [inline src/app/navbar/navbar.component.css]: w-100

File: src/app/navbar/navbar.component.ts [hunk 5]
```diff
@@ -55,6 +56,7 @@ export class NavbarComponent extends Logout implements OnInit {
 
   ngOnInit() {
     this.userService.user$.subscribe(user => this.user = user);
+    this.userService.extendedUser$.subscribe(extendedUser => this.extendedUser = extendedUser);
```
Comment (garyluu) [inline src/app/navbar/navbar.component.ts]: ngUnsubscribe, the other one too while you're at it.

File: src/app/register/register.service.ts [hunk 6]
```diff
@@ -13,10 +13,12 @@
  *    See the License for the specific language governing permissions and
  *    limitations under the License.
  */
+
 import { Injectable } from '@angular/core';
-import { MatSnackBar } from '@angular/material';
-import { AuthService } from 'ng2-ui-auth';
 import { Observable } from 'rxjs';
+import { Response } from '@angular/http';
```
Comment (garyluu) [inline src/app/register/register.service.ts]: unused import?

File: src/app/loginComponents/accounts/internal/change-username/change-username.component.ts [hunk 7]
```diff
@@ -0,0 +1,80 @@
...
+  constructor(private userService: UserService, private usersService: UsersService, private refreshService: RefreshService) { }
+
+  ngOnInit() {
+    this.userService.user$.subscribe(user => {
+      if (user) {
+        this.user = user;
+        this.username = user.username;
+      }
+    });
+    this.userService.extendedUser$.subscribe(extendedUser => this.extendedUser = extendedUser);
```
Comment (garyluu) [inline src/app/loginComponents/accounts/internal/change-username/change-username.component.ts]: unsubscribe from anything that isn't an HTTP service

**Pull request merged**